File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AlwaysSetDateTimeKind.cs
using System;
using System.Globalization;
using MyAlias = System.DateTime;

public class Program
{
    public void Noncompliant()
    {
        var dt = new DateTime(); // Noncompliant {{Provide the "DateTimeKind" when creating this object.}}
        //       ^^^^^^^^^^^^^^
        dt = new DateTime(1623); // Noncompliant
        dt = new DateTime(1994, 07, 05); // Noncompliant
        dt = new DateTime(1994, 07, 05, new GregorianCalendar()); // Noncompliant
        dt = new DateTime(1994, 07, 05, 16, 23, 00); // Noncompliant
        dt = new DateTime(1994, 07, 05, 16, 23, 00, new GregorianCalendar()); // Noncompliant
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42); // Noncompliant
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42, new GregorianCalendar()); // Noncompliant
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42, new GregorianCalendar()); // Noncompliant
        dt = new MyAlias(); // FN
        dt = new System.DateTime(); // Noncompliant
    }

    public void Compliant()
    {
        var dt = new DateTime(1623, DateTimeKind.Unspecified);
        dt = new DateTime(1994, 07, 05, 16, 23, 00, DateTimeKind.Local);
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42, new GregorianCalendar(), DateTimeKind.Unspecified);
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42, DateTimeKind.Utc);
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42, new GregorianCalendar(), DateTimeKind.Unspecified);
        dt = new DateTime(1994, 07, 05, 16, 23, 00, 42, DateTimeKind.Unspecified);
        dt = new System.(1623); // Error [CS1001]
    }
}

public class FakeDateTime
{
    private class DateTime
    {

    }

    private void Compliant()
    {
        var dt = new DateTime();
    }
}

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AlwaysSetDateTimeKind.vb
Imports System.Globalization
Imports MyAlias = System.DateTime

Public Class Program

    Public Sub Noncompliant()
        Dim dt = New DateTime() ' Noncompliant {{Provide the "DateTimeKind" when creating this object.}}
        '        ^^^^^^^^^^^^^^
        dt = New Date() ' Noncompliant
        dt = New dATEtIME() ' Noncompliant
        dt = New DateTime(1623) ' Noncompliant
        dt = New DateTime(1994, 7, 5) ' Noncompliant
        dt = New DateTime(1994, 7, 5, New GregorianCalendar()) ' Noncompliant
        dt = New DateTime(1994, 7, 5, 16, 23, 0) ' Noncompliant
        dt = New DateTime(1994, 7, 5, 16, 23, 0, New GregorianCalendar()) ' Noncompliant
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42) ' Noncompliant
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42, New GregorianCalendar()) ' Noncompliant
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42, New GregorianCalendar()) ' Noncompliant
        dt = New MyAlias() ' FN
        dt = New System.DateTime() ' Noncompliant
    End Sub

    Public Sub Compliant()
        Dim dt = New DateTime(1623, DateTimeKind.Unspecified)
        dt = New DateTime(1994, 7, 5, 16, 23, 0, DateTimeKind.Local)
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42, New GregorianCalendar(), DateTimeKind.Unspecified)
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42, DateTimeKind.Utc)
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42, New GregorianCalendar(), DateTimeKind.Unspecified)
        dt = New DateTime(1994, 7, 5, 16, 23, 0, 42, DateTimeKind.Unspecified)
    End Sub

End Class

Class FakeDateTime

    Private Class DateTime

    End Class

    Private Sub Compliant()
        Dim dt = New DateTime()
    End Sub

End Class
